crates/amaru-kernel/src/protocol_parameters.rs

```diff
@@ -96,14 +97,25 @@ impl<'b, C> cbor::decode::Decode<'b, C> for ProtocolParameters {
 
         let coins_per_utxo_byte = d.u64()?;
 
-        let _ = d.map()?;
-        d.u8()?;
-        let plutus_v1 = d.decode_with(ctx)?;
-        d.u8()?;
-        let plutus_v2 = d.decode_with(ctx)?;
-        d.u8()?;
-        let plutus_v3 = d.decode_with(ctx)?;
-
+        let mut plutus_v1 = None;
+        let mut plutus_v2 = None;
+        let mut plutus_v3 = None;
+        let i = d.map_iter_with::<C, u8, CostModel>(ctx)?;
+        for item in i {
+            let (k, v) = item?;
+            match k {
+                0 => {
+                    plutus_v1 = Some(v);
+                }
+                1 => {
+                    plutus_v2 = Some(v);
+                }
+                2 => {
+                    plutus_v3 = Some(v);
+                }
+                _ => {}
```

Comment:
```suggestion
                _ => unreachable!("unexpected language version: {k}"),
```

Avoid silently swallowing unexpected behaviors.